### PR TITLE
OPT support DS 1x Inference

### PR DESCRIPTION
**Pull request by ZhaiFeiyue**
# What does this PR do?

facebook/opt-125m

ds mode

cmd 

```bash
python ../gaudi_spawn.py --use_deepspeed --world_size 1 run_generation.py --model_name_or_path facebook/opt-125m --batch_size 1 --use_hpu_graphs --use_kv_cache --max_new_tokens 100
```

outputs
image

non-ds mode

cmd

```bash
python run_generation.py --model_name_or_path facebook/opt-125m --batch_size 1 --use_hpu_graphs --use_kv_cache --max_new_tokens 100
```

outputs
image

know issue

- only support DS 1x, not support DS multi cards
- reasonable but different outputs of opt13B and bloom-560m between non-ds mode and ds mode, no sure if is a issue

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by ZhaiFeiyue** [comment]
@regisss with this PR OPT could run ds 1x inference, but ds multi card still not work, and also I can't run bloom-560m with ds 2X

File: examples/text-generation/checkpoint_utils.py [hunk 1]
```diff
@@ -66,13 +66,35 @@ def model_is_bloom(config):
     return "bloom" in config.architectures[0].lower()
 
 
+def get_optimized_model_name(config):
+    model_names = ["bloom", "gpt2", "opt", "gptj", "gptneox"]
+    for model_name in model_names:
+        if model_name in config.architectures[0].lower():
```
Comment (regisss) [inline examples/text-generation/checkpoint_utils.py:72]:
```suggestion
    model_names = ["bloom", "gpt2", "opt", "gptj", "gpt_neox"]
    for model_name in model_names:
        if model_name == config.model_type:
```
My previous implementation was not very clean, this is better XD

**Review by regisss — COMMENTED** [review]
I left just one comment which is mainly my fault XD

Also, what is the command line you used for BLOOM-560m DS 2x? I'm going to run it to see if I can fix it.

**Comment by ZhaiFeiyue** [comment]
> I left just one comment which is mainly my fault XD
> 
> Also, what is the command line you used for BLOOM-560m DS 2x? I'm going to run it to see if I can fix it.

my cmd is 

```bash
python ../gaudi_spawn.py --use_deepspeed --world_size 2 run_generation.py --model_name_or_path bigscience/bloom-560m --batch_size 1 --use_hpu_graphs --use_kv_cache --max_new_tokens 100
```

the error is 
image

i'm using R1.10, will try 1.9.0 later, no sure if is the env issue

File: examples/text-generation/checkpoint_utils.py (hunk 1 above)
Comment (ZhaiFeiyue) [reply examples/text-generation/checkpoint_utils.py]: done

**Comment by ZhaiFeiyue** [comment]
@regisss I will submit a another one to support GPT2/GPT_NeoX/GPT-J after this PR merged

**Comment by ZhaiFeiyue** [comment]
> > I left just one comment which is mainly my fault XD
> > Also, what is the command line you used for BLOOM-560m DS 2x? I'm going to run it to see if I can fix it.
> 
> my cmd is
> 
> ```shell
> python ../gaudi_spawn.py --use_deepspeed --world_size 2 run_generation.py --model_name_or_path bigscience/bloom-560m --batch_size 1 --use_hpu_graphs --use_kv_cache --max_new_tokens 100
> ```
> 
> the error is image
> 
> i'm using R1.10, will try 1.9.0 later, no sure if is the env issue

@regisss  this issue after debug is duce the flags not work
```python
os.environ.setdefault("PT_HPU_LAZY_ACC_PAR_MODE", "0")
os.environ.setdefault("PT_HPU_ENABLE_LAZY_COLLECTIVES", "true")
```
maybe is because environ set value after habana library loaded, please have a look and check if you have any ideas!

**Comment by ZhaiFeiyue** [comment]
@regisss  I will try OPT-125m with DS 2X and HPUGraph

**Comment by ZhaiFeiyue** [comment]
@regisss  with your patch, graph mode works well
image

**Review by regisss — APPROVED** [review]
LGTM!

**Pull request merged**